src/main/java/com/synopsys/integration/jira/common/rest/service/PluginManagerService.java

```diff
@@ -134,8 +134,13 @@ public String retrievePluginToken() throws IntegrationException {
         requestBuilder.addQueryParameter(QUERY_KEY_OS_AUTH_TYPE, QUERY_VALUE_OS_AUTH_TYPE);
         requestBuilder.method(HttpMethod.GET);
         requestBuilder.addHeader(ACCEPT_HEADER, MEDIA_TYPE_INSTALLED);
-        Map<String, String> response = jiraApiClient.getResponseHeaders(requestBuilder.build());
-        return response.get("upm-token");
+        JiraRequest request = requestBuilder.build();
+        Map<String, String> response = jiraApiClient.getResponseHeaders(request);
+        // Jira Cloud had a breaking change in its public API where the "upm-token" was renamed for their credential client
+        //  This change will support the new header name but keep the functionality of the old headers which are still used
+        //  by the OAuth client and Jira Server.
+        Optional<String> pluginToken = Optional.ofNullable(response.get("upm-token"));
+        return pluginToken.orElseGet(() -> response.get("Upm-Token"));
```

Comment:
```suggestion
        return Optional.ofNullable(response.get("upm-token"))
                                     .or(() -> Optional.ofNullable(response.get("Upm-Token")));
```
Do the client's of that use this method handle a potential null string?  Could we change the method signature to return an Optional<String> but we would need to change the major version of the library since that would be a breaking change.